### PR TITLE
Add datasource to coverage

### PR DESCRIPTION
**Pull request by datanel**
- Added new API POST /coverages/:id/data_sources
```json
{ "id": "8484-14545-14484" }
```

- GET /coverages/:id, POST /coverages, PATCH /coverages/:id now return an array of coverages
```json
{
  "coverages": [
    {
      ....
    }
  ]
}
```

Before
```json
{
  "coverages": {
      ...
  }
}
```

- DELETE /coverages/:id now return nothing

### CODE REVIEW

File: documentation/openAPI.yaml [hunk 1]
```diff
@@ -164,6 +164,37 @@ paths:
           schema:
             $ref: '#/definitions/message'
 
+  /coverages/{coverage_id}/data_sources:
+    post:
+      summary: Add a data_source to coverage <coverage_id>
+      tags:
+        - coverages
```
Comment (datanel) [inline documentation/openAPI.yaml]: this api only concerns coverages

File: documentation/openAPI.yaml [hunk 2]
```diff
@@ -164,6 +164,37 @@ paths:
           schema:
             $ref: '#/definitions/message'
 
+  /coverages/{coverage_id}/data_sources:
+    post:
+      summary: Add a data_source to coverage <coverage_id>
```
Comment (datanel) [inline documentation/openAPI.yaml]: not sure this will make a real reference

File: documentation/openAPI.yaml [hunk 3]
```diff
@@ -411,6 +446,13 @@ definitions:
       environments:
         $ref: "#/definitions/environments"
 
+  data_source_id:
+    type: object
+    properties:
+      id:
+        type: string
+        example: 5810cb2d3c96b7000108d074
```
Comment (pbougue) [inline documentation/openAPI.yaml]: clearer example might be nice (like `lyon_tram`) but it's also the case for previous doc

File: tartare/api.py [hunk 4]
```diff
@@ -53,3 +54,4 @@
                              endpoint='data')
 api.add_resource(Contributor, '/contributors', '/contributors/', '/contributors/<string:contributor_id>', endpoint='contributors')
 api.add_resource(DataSource, '/contributors/<string:contributor_id>/data_sources', '/contributors/<string:contributor_id>/data_sources/<string:data_source_id>')
+api.add_resource(CoverageDataSourceSubscription, '/coverages/<string:coverage_id>/data_sources', '/coverages/<string:coverage_id>/data_sources/<string:data_source_id>', endpoint='coverages_data_sources')
```
Comment (pbougue) [inline tartare/api.py]: the way I understood it, endpoint attribute is just to present it on root `/ `, so I wouldn't let it here

File: tartare/interfaces/coverage_data_source_subscription.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        data_source_id = request.json['id']
+
+        try:
+            data_sources = models.DataSource.get(data_source_id=data_source_id)
+        except ValueError:
+            return {'message': 'Unknown data_source_id {}.'.format(data_source_id, coverage_id)}, 400
+
+        if coverage.has_data_source(data_sources[0]):
+            return {'message': 'Data source id {} already exists in coverage {}.'.format(data_source_id, coverage_id)}, 400
```
Comment (pbougue) [inline tartare/interfaces/coverage_data_source_subscription.py]: maybe more a 409 on that (not really a problem, or not the same as 400 above)

File: documentation/openAPI.yaml (hunk 3 above)
Comment (datanel) [inline documentation/openAPI.yaml]: i will make another pr for that

**Pull request merged**